hw3/hw3.playground/Contents.swift

```diff
@@ -0,0 +1,464 @@
+        sedanCarParking.append(car)
+    }
+    
+    mutating func addCar(car:SportCar){//Static(Direct) Dispatch
+        sportCarParking.append(car)
+    }
+    
+    mutating func addCar(car: VanCar){//Static(Direct) Dispatch
+        vanCarParking.append(car)
+    }
+    
+    mutating func addCar(car:CrossOverCar){//Static(Direct) Dispatch
+        crossOverCarParking.append(car)
+    }
+    
+}
+
+public class Diller:Dillerable{
+    
+    public var sedanCarParking: [SedanCar] = []
+    
+    public var sportCarParking: [SportCar] = []
+    
+    public var vanCarParking: [VanCar] = []
+    
+    public var crossOverCarParking: [CrossOverCar] = []
+    
+    init(){
+    }
+}
+
+public class CarFactory{
+    private var name:String
+    var diller: Diller?
+    init(name: String, diller:Diller){
+        self.name = name
+        self.diller = diller
+    }
+    
+    func buildCar(typeCar: TypeCar, country: Location, manufacture: Manufacture) {
+        switch typeCar {
+        case .Sedan:
+            diller?.addCar(car: buildSedanCar(country: country, manufacture: manufacture, color: UIColor.blue))
+        case .Sport:
+            diller?.addCar(car: buildSportCar(country: country, manufacture: manufacture, color: UIColor.blue))
+        case .Van:
+            diller?.addCar(car: buildVanCar(country: country, manufacture: manufacture, color: UIColor.blue))
+        case .CrossOver:
+            diller?.addCar(car: buildCrossOverCar(country: country, manufacture: manufacture, color: UIColor.blue))
+        }
+    }
+    
+    func sellByVin(vin:String){
+        diller?.sellByVin(vin: vin)
+    }
+    func generateVin() -> String{
+        var vin = String()
+        let letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+        for _ in 0...16 {
+            vin += String(Array(letters)[Int.random(in: 0..<letters.count)])
+        }
+        return vin
+    }
+    
+    private func buildSedanCar(country: Location, manufacture: Manufacture, color: UIColor) -> SedanCar {
+
+        let car = SedanCar(country: country, manufacture: manufacture, vin: generateVin(), factory: self, color: color)
+        return car
+    }
+    
+    private func buildSportCar(country: Location, manufacture: Manufacture, color: UIColor) -> SportCar {
+        let car = SportCar(country: country, manufacture: manufacture, vin: generateVin(), factory: self, color: color)
+        return car
+    }
+    
+    private func buildVanCar(country: Location, manufacture: Manufacture, color: UIColor) -> VanCar {
+        let car = VanCar(country: country, manufacture: manufacture, vin: generateVin(), factory: self, color: color)
+        return car
+    }
+    
+    private func buildCrossOverCar(country: Location, manufacture: Manufacture, color: UIColor) -> CrossOverCar {
+        let car = CrossOverCar(country: country, manufacture: manufacture, vin: generateVin(), factory: self, color: color)
+        return car
+    }
+
+}
+
+var d = Diller()
```

Comment:
Нейминги должны быть полными
```suggestion
var diller = Diller()
```